moveit_core/robot_state/include/moveit/robot_state/robot_state.h

```diff
@@ -1699,6 +1699,27 @@ class RobotState
     attachBody(id, pose, shapes, shape_poses, touch_links_set, link_name, detach_posture, subframe_poses);
   }
 
+  [[deprecated("No offset pose for the attached shapes can be specified with this overload; using a default offset of "
+               "identity")]] void
```

Comment:
```suggestion
  [[deprecated("Use the new method that allows passing a base pose for the attached body. Using identity.")]] void
```

Reply:
Given this incompatibility, I agree with @v4hn and am inclined to reject this PR.

Reply:
That's fine with me. The version macro solution works well enough for my application